libass/ass_parse.c

```diff
@@ -996,22 +996,17 @@ void process_karaoke_effects(ASS_Renderer *render_priv)
             x_end = FFMAX(x_end, d6_to_int(info->bbox.x_max + info->pos.x));
         }
 
+        if (effect_type != EF_KARAOKE_KF)
+            tm_end = tm_start;
+
         int x;
-        if (effect_type == EF_KARAOKE || effect_type == EF_KARAOKE_KO) {
-            x = tm_current < tm_start ? x_start : x_end + 1;
-        } else if (effect_type == EF_KARAOKE_KF) {
-            if (tm_current < tm_start)
-                x = x_start;
-            else if (tm_current >= tm_end)
-                x = x_end + 1;
-            else {
-                double dt = (double) (tm_current - tm_start) / (tm_end - tm_start);
-                x = x_start + (x_end - x_start) * dt;
-            }
-        } else {
-            ass_msg(render_priv->library, MSGL_ERR,
```

Comment:
Is it impossible to reach here? Maybe it's better to add the corresponding assert somewhere?

Reply:
The Effect enum has no other values. We skip `EF_NONE` above, and we handle all other values here. This `ass_msg` could have been an `assert` all along. I don’t see any value in keeping it.